doc/string/aref.rdoc

```diff
@@ -0,0 +1,107 @@
+Returns the substring of +self+ specified by the arguments.
+
+Note: The form <tt>self[regexp, capture = 0]</tt>
+updates {Regexp-related global variables}[rdoc-ref:Regexp@Global+Variables];
+the other forms do not.
```

Comment:
We already say this in the "Form self[regexp, capture = 0]` section. I think it's a bit redundant to mention this twice since it's not that important anyways.